mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/social/StatefulFollowingScreen.kt

```diff
@@ -1,18 +1,29 @@
 package ch.epfl.sdp.mobile.ui.features.social
 
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
 import ch.epfl.sdp.mobile.data.api.AuthenticationApi
+import ch.epfl.sdp.mobile.data.api.firebase.FirebaseAuthenticatedUser
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
 
-// TODO : Implement this.
-@Suppress("Unused")
+@Preview
 @Composable
 fun StatefulFollowingScreen(
-    user: AuthenticationApi.User.Authenticated,
-    modifier: Modifier = Modifier,
+        user: AuthenticationApi.User.Authenticated,
+        modifier: Modifier = Modifier,
 ) {
-  Box(modifier, Alignment.Center) { Text("Social") }
+  val following by user.following.collectAsState(initial = emptyList())
+
+  Box(modifier, Alignment.Center) { following.forEach { profile ->  Text(profile.name, modifier = Modifier.testTag("following")) } }
 }
```

Comment:
I believe your PR should not really be anything more than this :

```kotlin
package ch.epfl.sdp.mobile.ui.features.social

import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.runtime.State
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import ch.epfl.sdp.mobile.data.api.AuthenticationApi
import ch.epfl.sdp.mobile.data.api.ProfileColor

private class SnapshotFollowingState(
    private val following: State<List<AuthenticationApi.Profile>>,
) : FollowingState {

  private data class ProfileAdapter(
      private val profile: AuthenticationApi.Profile,
  ) : Person {
    override val backgroundColor: ProfileColor
      get() = profile.backgroundColor
    override val name: String
      get() = profile.name
    override val emoji: String
      get() = profile.emoji
  }

  override val players: List<Person>
    get() = following.value.map { ProfileAdapter(it) }
}

@Composable
fun StatefulFollowingScreen(
    user: AuthenticationApi.User.Authenticated,
    modifier: Modifier = Modifier,
) {
  val state = remember(user) { user.following }.collectAsState(emptyList())
  SocialScreen(SnapshotFollowingState(state), modifier.fillMaxSize())
}
```

It may not a few integration tests, but not much more is needed.